ls1intum/Artemis
# `Development`: Add Aeolus build plan templates for Haskell for local continuous integration

### Checklist
#### General


#### Server

### Motivation and Context


This PR adds the templates for build plans for the language Haskell. 

### Description

I added the templates and added the `ProgrammingLanguageFeature` for LocalCI so one can now create custom build plans (starting with the help of the template) for Haskell in LocalCI and Bamboo.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Students
- Programming Exercise Setup with either LocalCI or Bamboo

1. Log in to Artemis
2. Create a new Haskell exercise and enable custom build plans
3. Confirm that the build plan behaves the same as it does in Bamboo. meaning that base fails and solution hopefully passes

#### Exam Mode Testing



Prerequisites:
- 1 Instructor
- 2 Students
- 1 Exam with a Programming Exercise

1. Log in to Artemis
2. Create an exam
3. Create a new C exercise and enable custom build plans
4. Confirm that the build plan behaves the same as it does in Bamboo. meaning that base fails and solution hopefully passes

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)
